### PR TITLE
Removes bootstrap dependency

### PR DESCRIPTION
**Pull request by joao-f-medeiros**
Issue [here](https://github.com/OHIF/react-viewerbase/issues/52).

### CODE REVIEW

**Comment by dannyrb** [comment]
### Measurements Table

image

### Quick Switch

Should show a dark background fade-in when hovered. You can more clearly see the difference when testing the previous Netlify deploy: https://deploy-preview-97--ohif-react-viewerbase.netlify.com/components/quick-switch

### Table List

Had a wider area for the index in PR 97

image

### User Preferences Modal

Does not open

**Comment by dannyrb** [comment]
Pulling this down to test locally :+1:

**Comment by dannyrb** [comment]
Looks great in `Viewers`. A couple of issues with modals.

### User Preferences

- Buttons are missing styles
- Gutter/container margins/padding appear to be off. Note how close labels and textboxes are to the edge of the modal

firefox_JksMtrlOn5

### About Modal

This is not yet live. So I'm okay to merge before it's fixed, but worth noting that this appears to rely on missing styles. CC: @biharck 

firefox_rDQ6ehZ80Z

File: src/__docs__/index.html [hunk 1]
```diff
@@ -67,6 +66,37 @@
...
+    .row {
+      margin-right: -15px;
+      margin-left: -15px;
+    }
+
+    * {
+      -webkit-box-sizing: border-box;
+      -moz-box-sizing: border-box;
+      box-sizing: border-box;
+    }
```
Comment (dannyrb) [inline src/__docs__/index.html]: Do these styles need to be bundled with our output? Or are they just resets to protect us from docz values with our components?

File: src/components/layoutButton/LayoutChooser.styl [hunk 2]
```diff
@@ -1,5 +1,34 @@
 $borderColor = rgba(77, 99, 110, 0.81)
 
+.pull-left
+  float: left!important;
```
Comment (dannyrb) [inline src/components/layoutButton/LayoutChooser.styl]: Are imported react styles like this scoped to the component/module, or do they bleed out? I'm unfamiliar here.

If they bleed out, this should be moved to a utility class file somewhere else.

File: src/components/layoutButton/LayoutChooser.styl [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+  border: 1px solid #ccc;
+  border: 1px solid rgba(0,0,0,.15);
+  border-radius: 4px;
+  -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
+  box-shadow: 0 6px 12px rgba(0,0,0,.175);
+
+.btn-group, .btn-group-vertical
+  position: relative;
+  display: inline-block;
+  vertical-align: middle;
```
Comment (dannyrb) [inline src/components/layoutButton/LayoutChooser.styl]: Similarly, `.btn-group` may be defined "higher up", or included in a button specific file

File: src/components/quickSwitch/QuickSwitch.styl [hunk 4]
```diff
@@ -5,12 +5,15 @@
...
     opacity: 0
+    content: ''
 
   &:hover
     z-index: 50
 
     &:before
+      display: table;
       opacity: 1
+      content: ''
```
Comment (dannyrb) [inline src/components/quickSwitch/QuickSwitch.styl:16]: Are we using this as a flexbox alternative?

File: src/components/studyList/StudyList.styl [hunk 5]
```diff
@@ -49,6 +49,9 @@ placeholder-color(c)
     font-weight: 100
     line-height: $study-list-toolbar-height
 
+  .pull-right
+    float: right!important;
```
Comment (dannyrb) [inline src/components/studyList/StudyList.styl]: Same for this utility class

File: src/components/studyList/StudyList.styl [hunk 6]
```diff
@@ -194,6 +205,40 @@ placeholder-color(c)
             border-bottom: 1px solid var(--ui-gray-lighter)
             background-color: var(--table-hover-color)
 
+.form-control
```
Comment (dannyrb) [inline src/components/studyList/StudyList.styl]: Form-control should probably be in a more generic `.styl` file

File: src/components/tooltip/Tooltip.styl [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+.tooltip-arrow {
+  position: absolute;
+  width: 0;
+  height: 0;
+  border-color: transparent;
+  border-style: solid;
+}
+
+.fade
+  opacity: 0;
```
Comment (dannyrb) [inline src/components/tooltip/Tooltip.styl]: Should probably be in a more generic transitions file

File: src/components/userPreferencesModal/HotKeysPreferences.styl [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.col-md-6
```
Comment (dannyrb) [inline src/components/userPreferencesModal/HotKeysPreferences.styl]: You may want to rip this out and update the preferences modal to use flexbox

File: src/components/userPreferencesModal/UserPreferencesModal.styl [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    display: block;
+
+  > li
+    float: left;
+    margin-bottom: -1px;
+
+.modal-open
+  overflow: hidden;
+
+.modal
```
Comment (dannyrb) [inline src/components/userPreferencesModal/UserPreferencesModal.styl]: Do we have a more generic "modal" component that should contain these?

File: src/components/userPreferencesModal/UserPreferencesModal.styl [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  color: #000;
+  text-shadow: 0 1px 0 #fff;
+  opacity: .2;
+
+.row
+  margin-right: -15px;
+  margin-left: -15px;
+
+.pull-left
+  float: left!important;
```
Comment (dannyrb) [inline src/components/userPreferencesModal/UserPreferencesModal.styl]: Ah. See. No reason to duplicate this `.pull-left`

File: src/components/userPreferencesModal/UserPreferencesModal.styl [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+ol, ul
+  margin-top: 0;
```
Comment (dannyrb) [inline src/components/userPreferencesModal/UserPreferencesModal.styl]: Again, unless these are scoped, things like this feel dangerous

**Review by dannyrb — CHANGES_REQUESTED** [review]
Overall, I think this does a good job of achieving the desired effect. I think we mostly need to move things around and co-locate re-used bits of code that are more global.

File: src/__docs__/index.html (hunk 1 above)
Comment (joao-f-medeiros) [reply src/__docs__/index.html]: Good question, if we need to guarantee these components to be totally separated from bootstrap dependency, we have to have these styles in the bundle.

File: src/components/layoutButton/LayoutChooser.styl (hunk 2 above)
Comment (joao-f-medeiros) [reply src/components/layoutButton/LayoutChooser.styl]: They can bleed out. I will change to be scoped only in this component

File: src/components/layoutButton/LayoutChooser.styl (hunk 3 above)
Comment (joao-f-medeiros) [reply src/components/layoutButton/LayoutChooser.styl]: Ok

File: src/components/quickSwitch/QuickSwitch.styl (hunk 4 above)
Comment (joao-f-medeiros) [reply src/components/quickSwitch/QuickSwitch.styl:16]: I added the `content` property here to fix the problem where the `before` was not being rendered after removing the bootstrap. This makes `before` legible to be presented.

File: src/components/studyList/StudyList.styl (hunk 5 above)
Comment (joao-f-medeiros) [reply src/components/studyList/StudyList.styl]: Ok, I will change

File: src/components/studyList/StudyList.styl (hunk 6 above)
Comment (joao-f-medeiros) [reply src/components/studyList/StudyList.styl]: Form-control is being used only in `StudyList` component, should we move this anyway?

Comment (dannyrb) [reply src/components/studyList/StudyList.styl]: Just to be safe, I would appreciate it if it were part of some styles at a higher level up. It'll make it easier to notice/change as we pull in and style form components as "elements"

File: src/components/tooltip/Tooltip.styl (hunk 7 above)
Comment (joao-f-medeiros) [reply src/components/tooltip/Tooltip.styl]: Ok, I will move to another file

File: src/components/userPreferencesModal/HotKeysPreferences.styl (hunk 8 above)
Comment (joao-f-medeiros) [reply src/components/userPreferencesModal/HotKeysPreferences.styl]: Ok

File: src/components/userPreferencesModal/UserPreferencesModal.styl (hunk 10 above)
Comment (joao-f-medeiros) [reply src/components/userPreferencesModal/UserPreferencesModal.styl]: Yeah, I will move all to the same place.

File: src/components/studyList/StudyList.styl (hunk 6 above)
Comment (joao-f-medeiros) [reply src/components/studyList/StudyList.styl]: Ok, I will move it

File: src/components/userPreferencesModal/UserPreferencesModal.styl (hunk 11 above)
Comment (joao-f-medeiros) [reply src/components/userPreferencesModal/UserPreferencesModal.styl]: I will change to keep component scoped

File: src/components/userPreferencesModal/UserPreferencesModal.styl (hunk 9 above)
Comment (joao-f-medeiros) [reply src/components/userPreferencesModal/UserPreferencesModal.styl]: I didn't find it, we can create a more generic style file for modal css and use it here, what do you think?

**Comment by swederik** [comment]
https://deploy-preview-100--ohif-react-viewerbase.netlify.com/components/toolbar-section

`TypeError: Cannot read property 'includes' of undefined`

Edit: looks like this is in master...

**Review by dannyrb — CHANGES_REQUESTED** [review]
### Simple Dialog

image

- Button styles are not applied
- Top right "close" looks off
- Input fields are using inconsistent styling
- Worth noting that the `UserPreferencesModal` uses a different "close button" and button styles do show correctly


### Table List

image

- Spacing between "block" and text should match spacing in measurement's table component

**Pull request merged**